js/app.js

```diff
@@ -3,37 +3,250 @@ function Calculator() {
     this.history = [];
 }
 
-Calculator.prototype.isCorrectAction = function(action) {
+Calculator.prototype.isCorrectAction = function (action) {
     return this.actions.includes(action);
 }
 
-Calculator.prototype.getHistoryAsString = function() {
+Calculator.prototype.getHistoryAsString = function () {
     return this.history.join('\n');
 }
 
-Calculator.prototype.add = function(num1, num2) {
+Calculator.prototype._calculate = function(num1, num2, operator, operation) {
+    const a = Number(num1);
+    const b = Number(num2);
+
+    if (isNaN(a) || isNaN(b)) { // spawdza czy wartości to liczby 
+        console.error("Wpisz liczby!");
+        return null;
+    }
+
+    if (operator === '/' && b === 0) { // b nie moze byc === 0
+        console.error("Pamiętaj cholero nie dziel przez ZERO!");
+        return null;
+    }
```

Comment:
Podejście do refaktoryzacji ok, ale ma to pewne wady - jeśli będziemy mieli 100 różnych działań to pomyśl jak bardzo ta funkcja się rozrośnie - my tego nie chcemy. Chcemy kod dzielić na małe kawałki (funkcje). Dlatego pozostawienie tego w poszczególnej funkcji będzie lepszym rozwiązaniem. Samo parsowanie liczb można wrzucić do linii 235 tj.
```
number1 = prompt('Podaj liczbę nr 1');
number2 = prompt('Podaj liczbę nr 2');
```

Najważniejsze tutaj jest zmiana podejścia do "uruchamiania" samej funkcji. Warto zrobić odpowiednią "strukturę danych". Jak? Ten artykuł Ci pomoże: https://devmentor.pl/b/mniej-instrukcji-warunkowych